# Group project for COMP2006

Group members:
- Cecilia Dias Cardoso Da Mota
- Nicolas Trillo Gutierrez

In [1]:
# Google pandas import
import pandas as pd

# Reading the datasets
red_wine_quality = pd.read_csv('./data/wine-quality/winequality-red.csv', sep=';')
white_wine_quality = pd.read_csv('./data/wine-quality/winequality-white.csv', sep=';')
wine_chemistry = pd.read_csv('./data/wine/wine.data', sep=',', header=None, names=[
    "Alcohol", "Malic acid", "Ash", "Alkalinity of ash", "Magnesium", 
    "Total phenols", "Flavonoids", "Nonflavonoid phenols", "Proanthocyanins", 
    "Colour intensity", "Hue", "OD280/OD315 of diluted wines", "Proline"
])

# Target Column Identification
# For the red wine dataset
red_target_col = 'quality'
red_target_type = "Numeric" if pd.api.types.is_numeric_dtype(red_wine_quality[red_target_col]) else "Categorical"

# For the white wine dataset
white_target_col = 'quality'
white_target_type = "Numeric" if pd.api.types.is_numeric_dtype(white_wine_quality[white_target_col]) else "Categorical"

# For the wine chemistry dataset
chemistry_target_col = 'Alcohol'
chemistry_target_type = "Numeric" if pd.api.types.is_numeric_dtype(wine_chemistry[chemistry_target_col]) else "Categorical"

# Displaying the results
print(f"Dataset: winequality-red.csv\nTarget Column: {red_target_col}\nData Type: {red_target_type}\n")
print(f"Dataset: winequality-white.csv\nTarget Column: {white_target_col}\nData Type: {white_target_type}\n")
print(f"Dataset: wine.data\nTarget Column: {chemistry_target_col}\nData Type: {chemistry_target_type}")


Dataset: winequality-red.csv
Target Column: quality
Data Type: Numeric

Dataset: winequality-white.csv
Target Column: quality
Data Type: Numeric

Dataset: wine.data
Target Column: Alcohol
Data Type: Numeric


### Red Wine Quality
Obtained from the UC Irvine ML data set repository, [Link](https://archive.ics.uci.edu/dataset/186/wine+quality)

In [2]:
red_wine_quality.info()
red_wine_quality.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1599 entries, 0 to 1598
Data columns (total 12 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   fixed acidity         1599 non-null   float64
 1   volatile acidity      1599 non-null   float64
 2   citric acid           1599 non-null   float64
 3   residual sugar        1599 non-null   float64
 4   chlorides             1599 non-null   float64
 5   free sulfur dioxide   1599 non-null   float64
 6   total sulfur dioxide  1599 non-null   float64
 7   density               1599 non-null   float64
 8   pH                    1599 non-null   float64
 9   sulphates             1599 non-null   float64
 10  alcohol               1599 non-null   float64
 11  quality               1599 non-null   int64  
dtypes: float64(11), int64(1)
memory usage: 150.0 KB


fixed acidity  volatile acidity  citric acid  residual sugar  chlorides  \
0            7.4              0.70         0.00             1.9      0.076   
1            7.8              0.88         0.00             2.6      0.098   
2            7.8              0.76         0.04             2.3      0.092   
3           11.2              0.28         0.56             1.9      0.075   
4            7.4              0.70         0.00             1.9      0.076   

   free sulfur dioxide  total sulfur dioxide  density    pH  sulphates  \
0                 11.0                  34.0   0.9978  3.51       0.56   
1                 25.0                  67.0   0.9968  3.20       0.68   
2                 15.0                  54.0   0.9970  3.26       0.65   
3                 17.0                  60.0   0.9980  3.16       0.58   
4                 11.0                  34.0   0.9978  3.51       0.56   

   alcohol  quality  
0      9.4        5  
1      9.8        5  
2      9.8        5  
3      9.8        6  
4      9.4        5

### White Wine Quality
Obtained from the UC Irvine ML data set repository, [Link](https://archive.ics.uci.edu/dataset/186/wine+quality)

In [3]:
white_wine_quality.info()
white_wine_quality.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4898 entries, 0 to 4897
Data columns (total 12 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   fixed acidity         4898 non-null   float64
 1   volatile acidity      4898 non-null   float64
 2   citric acid           4898 non-null   float64
 3   residual sugar        4898 non-null   float64
 4   chlorides             4898 non-null   float64
 5   free sulfur dioxide   4898 non-null   float64
 6   total sulfur dioxide  4898 non-null   float64
 7   density               4898 non-null   float64
 8   pH                    4898 non-null   float64
 9   sulphates             4898 non-null   float64
 10  alcohol               4898 non-null   float64
 11  quality               4898 non-null   int64  
dtypes: float64(11), int64(1)
memory usage: 459.3 KB


fixed acidity  volatile acidity  citric acid  residual sugar  chlorides  \
0            7.0              0.27         0.36            20.7      0.045   
1            6.3              0.30         0.34             1.6      0.049   
2            8.1              0.28         0.40             6.9      0.050   
3            7.2              0.23         0.32             8.5      0.058   
4            7.2              0.23         0.32             8.5      0.058   

   free sulfur dioxide  total sulfur dioxide  density    pH  sulphates  \
0                 45.0                 170.0   1.0010  3.00       0.45   
1                 14.0                 132.0   0.9940  3.30       0.49   
2                 30.0                  97.0   0.9951  3.26       0.44   
3                 47.0                 186.0   0.9956  3.19       0.40   
4                 47.0                 186.0   0.9956  3.19       0.40   

   alcohol  quality  
0      8.8        6  
1      9.5        6  
2     10.1        6  
3      9.9        6  
4      9.9        6

### Wine dataset
"Using chemical analysis to determine the origin of wines" (Aeberhard, S. et al.)

Obtained from the UC Irvine ML data set repository, [Link](https://archive.ics.uci.edu/dataset/109/wine)

_Note:_ Table headers were manually added, as per the `wine.names` file provided in the dataset.

In [4]:
wine_chemistry.info()
wine_chemistry.head()

<class 'pandas.core.frame.DataFrame'>
Index: 178 entries, 1 to 3
Data columns (total 13 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   Alcohol                       178 non-null    float64
 1   Malic acid                    178 non-null    float64
 2   Ash                           178 non-null    float64
 3   Alkalinity of ash             178 non-null    float64
 4   Magnesium                     178 non-null    int64  
 5   Total phenols                 178 non-null    float64
 6   Flavonoids                    178 non-null    float64
 7   Nonflavonoid phenols          178 non-null    float64
 8   Proanthocyanins               178 non-null    float64
 9   Colour intensity              178 non-null    float64
 10  Hue                           178 non-null    float64
 11  OD280/OD315 of diluted wines  178 non-null    float64
 12  Proline                       178 non-null    int64  
dtypes: float64(1

Alcohol  Malic acid   Ash  Alkalinity of ash  Magnesium  Total phenols  \
1    14.23        1.71  2.43               15.6        127           2.80   
1    13.20        1.78  2.14               11.2        100           2.65   
1    13.16        2.36  2.67               18.6        101           2.80   
1    14.37        1.95  2.50               16.8        113           3.85   
1    13.24        2.59  2.87               21.0        118           2.80   

   Flavonoids  Nonflavonoid phenols  Proanthocyanins  Colour intensity   Hue  \
1        3.06                  0.28             2.29              5.64  1.04   
1        2.76                  0.26             1.28              4.38  1.05   
1        3.24                  0.30             2.81              5.68  1.03   
1        3.49                  0.24             2.18              7.80  0.86   
1        2.69                  0.39             1.82              4.32  1.04   

   OD280/OD315 of diluted wines  Proline  
1                          3.92     1065  
1                          3.40     1050  
1                          3.17     1185  
1                          3.45     1480  
1                          2.93      735

In [5]:
# Part 2
#
# None of the data sets have missing values.
# White Wine Quality was chosen for the next
# steps.

df_num = white_wine_quality
x_train = df_num.drop('pH', axis=1)
y_train = df_num['pH']

In [6]:
# Part 3

from sklearn.ensemble import RandomForestRegressor

rf = RandomForestRegressor(n_estimators=100, n_jobs=-1)
rf.fit(x_train, y_train)

RandomForestRegressor(n_jobs=-1)

In [7]:
r2 = rf.score(x_train, y_train)
print(f"{r2:.4f}")

0.9577


### Comment on R2 value
The RF Score on the White Wine Quality data set (0.9577) indicates a very strong relationship in the DS's values, this means other quality metrics for wines with incomplete values are able to be predicted with high confidence.